packages/wonder-blocks-birthday-picker/src/components/birthday-picker.js

```diff
@@ -0,0 +1,236 @@
+// @flow
+import moment from "moment";
+import * as React from "react";
+import {View} from "@khanacademy/wonder-blocks-core";
+import {Strut} from "@khanacademy/wonder-blocks-layout";
+import Spacing from "@khanacademy/wonder-blocks-spacing";
+import {Body} from "@khanacademy/wonder-blocks-typography";
+import Icon, {icons} from "@khanacademy/wonder-blocks-icon";
+import Color from "@khanacademy/wonder-blocks-color";
+import {SingleSelect, OptionItem} from "@khanacademy/wonder-blocks-dropdown";
+
+type Props = {|
+    /**
+     * The default value to populate the birthdate with. Should be in the
+     * format: YYYY-MM-DD (e.g. 2021-05-26). It's only used to populate the
+     * initial value as this is an uncontrolled component.
+     */
+    defaultValue?: string,
+
+    /**
+     * Listen for changes to the birthdate. Could be a string in the YYYY-MM-DD
+     * format or `null`.
+     */
+    onChange: (date: ?string) => mixed,
+|};
+
+type State = {|
+    month: string | null,
+    day: string | null,
+    year: string | null,
+    error: string | null,
```

Comment:
What do you think about commenting on these like we do props? I don't feel strongly about it, but perhaps it's useful to understand how the state changes reflect in the component behavior?